pine64/pinebook-pro/kernel/default.nix

```diff
@@ -0,0 +1,90 @@
+    "1007-drm-rockchip-add-support-for-modeline-32MHz-e.patch" # DP Alt Mode
+    "1008-rk3399-rp64-pcie-Reimplement-rockchip-PCIe-bus-scan-delay.patch" # RockPro64
+    "1010-arm64-dts-amlogic-add-initial-Beelink-GT1-Ultimate-dev.patch" # Beelink
+    "1011-arm64-dts-amlogic-add-meson-g12b-ugoos-am6-plus.patch" # Meson Ugoos
+    "1012-drm-panfrost-scheduler-improvements.patch" # Panfrost
+    "1013-arm64-dts-rockchip-Add-PCIe-bus-scan-delay-to-RockPr.patch" # RockPro64
+    "1014-drm-rockchip-support-gamma-control-on-RK3399.patch" # RK3399 VOP
+    "1015-media-rockchip-rga-do-proper-error-checking-in-probe.patch" # Rockchip RGA
+    "1016-arm64-dts-rockchip-switch-to-hs200-on-rockpi4.patch" # Radxa Rock Pi 4
+    "1017-arm64-dts-meson-remove-CPU-opps-below-1GHz-for-G12B-boards.patch" # AMLogic [1/2]
+    "1018-arm64-dts-meson-remove-CPU-opps-below-1GHz-for-SM1-boards.patch" # AMLogic [2/2]
+    "1019-arm64-dts-rockchip-Add-PCIe-bus-scan-delay-to-Rock-P.patch" # Radxa Rock Pi 4
+
+    # Assorted Pinebook, PinePhone and PineTab patches
+    "2001-Bluetooth-Add-new-quirk-for-broken-local-ext-features.patch" # Bluetooth
+    "2002-Bluetooth-btrtl-add-support-for-the-RTL8723CS.patch" # Bluetooth
+    "2003-arm64-allwinner-a64-enable-Bluetooth-On-Pinebook.patch" # Bluetooth
+    "2004-arm64-dts-allwinner-enable-bluetooth-pinetab-pinepho.patch" # Bluetooth
+    "2005-staging-add-rtl8723cs-driver.patch" # Realtek WiFi
+    "2006-arm64-dts-allwinner-pinetab-add-accelerometer.patch" # Accelerometer
+    "2007-arm64-dts-allwinner-pinetab-enable-jack-detection.patch" # Audio
+    "2008-brcmfmac-USB-probing-provides-no-board-type.patch" # Bluetooth
+    "2009-dts-rockchip-Adapt-and-adopt-Type-C-support-from-Pin.patch" # DP Alt Mode
+
+    # Pinebook Pro Type-C patches from megous; original patch numbers found
+    # on https://xff.cz/kernels/5.17/patches/ are retained, with just the first
+    # digit changed from 0 to 3, to make tracking easier
+    "3170-arm64-dts-rk3399-pinebook-pro-Fix-USB-PD-charging.patch"
+    "3172-arm64-dts-rk3399-pinebook-pro-Improve-Type-C-support.patch"
+    "3174-arm64-dts-rk3399-pinebook-pro-Remove-redundant-pinct.patch"
+    "3178-arm64-dts-rk3399-pinebook-pro-Don-t-allow-usb2-phy-d.patch"
+    "3339-drm-rockchip-cdn-dp-Disable-CDN-DP-on-disconnect.patch"
+    "3355-usb-typec-fusb302-Set-the-current-before-enabling-pu.patch"
+    "3359-usb-typec-fusb302-Update-VBUS-state-even-if-VBUS-int.patch"
+    "3361-usb-typec-fusb302-Add-OF-extcon-support.patch"
+    "3362-usb-typec-fusb302-Fix-register-definitions.patch"
+    "3363-usb-typec-fusb302-Clear-interrupts-before-we-start-t.patch"
+    "3364-usb-typec-typec-extcon-Add-typec-extcon-bridge-drive.patch"
+    "3365-phy-rockchip-typec-Make-sure-the-plug-orientation-is.patch"
+    "3372-phy-rockchip-inno-usb2-More-robust-charger-detection.patch"
+    "3373-usb-typec-extcon-Don-t-touch-charger-proprties.patch"
```

Comment:
Shouldn't we only include the RK3399/Pinebook Pro patches?

Reply:
I wouldn't mind if we could have one kernel in here for multiple devices...

Reply:
@samueldr this was my initial idea too, but people I talked with in the Pinebook Pro IRC chat told me to not worry and that applying the other patches is only useless, but not harmful. However I could try building it again using only a subset of patches (but I'm not sure whether to consider some or not).

@Mic92 this is a nice idea, if you prefer I could keep the kernel definition in another folder and import it in Pinebook Pro's configuration. But where would you put it? `common/kernel`?

Reply:
Yes, applying extra patches is generally "useless", that is right.

My thoughts here is that this needlessly adds surface area to review and validate, if users want to do so. Though I'm not the one to decide here, as I'll keep using the mainline kernel.